emcc.py

```diff
@@ -1565,8 +1565,8 @@ def get_bitcode_file(input_file):
           options.debug_level = 4
           shared.Settings.DEBUG_LEVEL = 4
 
-      if options.debug_level > 3:
-          newargs.append('-fno-discard-value-names')
+      if options.debug_level > 3 and not shared.Settings.WASM_BACKEND:
```

Comment:
I think you should check "not shared.Settings.WASM" instead.. as this is an asmjs-only thing AFAICT.

Reply:
Perhaps with a comment:  "Unlike other targets, asmjs can preserver the internal llvm names in the final output..."